chain/stmgr/execute.go

```diff
@@ -13,6 +13,15 @@ import (
 )
 
 func (sm *StateManager) TipSetState(ctx context.Context, ts *types.TipSet) (st cid.Cid, rec cid.Cid, err error) {
+	return sm.TipSetStateWithOpts(ctx, ts, &TipSetStateOpts{})
+}
+
+type TipSetStateOpts struct {
```

Comment:
I'm not convinced this is necessary? With the existing tipsetstate cache, and the newly added fast lookup in https://github.com/filecoin-project/lotus/pull/10445, I think we can drop this. We'll only wind up executing the tipset (making the write) if:

- we're executing a tipset that isn't in our main chain
- we're missing the actual state / receipts root.